libs/commons/Textedit.ml

```diff
@@ -95,12 +95,28 @@ let partition_edits_by_file edits =
     edits_by_file;
   edits_by_file
 
+(* To align the autofix logic with the Python CLI, remove the newline
+     character when replacing an entire line with an empty string.*)
+let remove_newline_for_empty_replacement text edit =
+  let text_len = String.length text in
+  let line_start = edit.start = 0 || String.get text (edit.start - 1) = '\n' in
+  let is_empty_replacement = String.length edit.replacement_text = 0 in
+  let skip_char c i =
+    if i + 1 <= text_len && String.get text i = c then i + 1 else i
+  in
+  if line_start && is_empty_replacement then
+    { edit with end_ = edit.end_ |> skip_char '\r' |> skip_char '\n' }
+  else edit
+
 let apply_edits ~dryrun edits =
   let edits_by_file = partition_edits_by_file edits in
   let all_conflicting_edits = ref [] in
   Hashtbl.iter
     (fun file file_edits ->
       let file_text = Common.read_file file in
+      let file_edits =
+        List.map (remove_newline_for_empty_replacement file_text) file_edits
```

Comment:
Not sure why our Semgrep rule didn't catch this, but we use `Common.map` because it's tail recursive on long lists so as to avoid stack overflows.

Reply:
Fixed. Thanks for the comment. It's interesting to note that `List.map` isn't tail-recursive 😮